offix/src/main/java/org/aerogear/offix/interceptor/ConflictInterceptor.kt

```diff
@@ -0,0 +1,35 @@
+package org.aerogear.offix.interceptor
+
+import android.util.Log
+import okhttp3.Interceptor
+import okhttp3.Response
+import org.aerogear.offix.detectedConflict
+import java.nio.charset.Charset
+
+/*
+ ConflictInterceptor should be used by the user while making an instance of apollo client
+ which will help in detecting conflicts.
+ */
+class ConflictInterceptor  : Interceptor{
+
+    override fun intercept(chain: Interceptor.Chain): Response {
+        val request = chain.request()
+        val response = chain.proceed(request)
+
+        //https://stackoverflow.com/a/33862068/10189663
+        val responseBody = response.body()
+        val bufferedSource = responseBody?.source()
+        bufferedSource?.request(Long.MAX_VALUE)
+        val buffer = bufferedSource?.buffer()
+        val responseBodyString = buffer?.clone()?.readString(Charset.forName("UTF-8")) ?: ""
+
+        Log.d("OffixClass", " Interceptor ** : $responseBodyString")
+
+        if (responseBodyString.contains("conflictInfo")) {
```

Comment:
This interceptor is attached to apollo client. So it would be called when any call is made using apollo client.

For now, it will intercept the request for queries and mutations both that would be made using the client. 

I'll check if apollo deals with errors or not. Also, yeah we can filter out for mutations only, maybe by reading the headers of the request, I'll check on it.